msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/InteractiveRequestParameters.java

```diff
@@ -130,4 +125,23 @@ public static InteractiveRequestParametersBuilder builder(URI redirectUri) {
         return builder()
                 .redirectUri(redirectUri);
     }
+
+    //This Builder class is used to override Lombok's default setter behavior for any fields defined in it
+    public static class InteractiveRequestParametersBuilder {
+
+        /**
+         * Sets the PopParameters for this request, allowing the request to retrieve proof-of-possession tokens rather than bearer tokens
+         *
+         * For more information, see {@link PopParameters} and https://aka.ms/msal4j-pop
+         *
+         * @param httpMethod a valid HTTP method, such as "GET" or "POST"
```

Comment:
Is there an enum you can use instead? Maybe https://docs.oracle.com/javame/8.0/api/httpclient/api/com/oracle/httpclient/HttpMethod.html ?

Reply:
I have no idea if this is a realistic scenario, but is there any reason to restrict this to just the common HTTP headers (like get and post) and not allow app developers to pass in custom HTTP headers?

I just tested MSALRuntime's `SetPopParams` API with an HTTP header of "ABCD" and it gave me back a brand new access token without issue. No idea if this token could actually be used to access something via an "ABCD" header, but it seems like MSALRuntime considered it to be valid input.

Reply:
Ack.

Note that this isn't an HTTP header, it's an HTTP method. The variability for methods is much smaller, but the spec does indeed say that new methods can be defined https://www.rfc-editor.org/rfc/rfc7231#section-4.3

Reply:
It is a great find. Even when a new method can be defined, I think we should just restrict it to the common values as this is what we support for now. Having httpMethod of type HttpMethod rather than String is very common in Java. I will incline towards making it accept HttpMethod too.

Reply:
When I said "custom HTTP headers" I meant "custom HTTP methods", but yeah I assume the use cases of that are pretty small. 

In the latest commit the parameter now takes in a value from our existing HTTPMethod enum rather than an arbitrary string.